.github/workflows/dependabot-auto-merge.yml

```diff
@@ -0,0 +1,41 @@
+name: Dependabot auto-merge
+on:
+  workflow_call:
+    inputs:
+      merge-option:
+        description: 'Merge option'
+        type: string
+        required: false
+        default: '--squash'
+    secrets:
+      token:
+        description: 'Token used to enable auto-merge and auto-approve'
+        required: true
+
+permissions:
+  pull-requests: write
+  contents: write
+
+jobs:
+  dependabot:
+    runs-on: ubuntu-latest
+    if: ${{ github.event.pull_request.user.login == 'dependabot[bot]' }}
+    steps:
+      - name: Dependabot metadata
+        id: metadata
+        uses: dependabot/fetch-metadata@v1.3.3
+        with:
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
+      - name: Enable auto-merge for minor and patch Dependabot PRs
+        if: ${{ steps.metadata.outputs.update-type == 'version-update:semver-minor' || steps.metadata.outputs.update-type == 'version-update:semver-patch' }}
+        run: gh pr merge --auto ${{ inputs.merge-option }} "$PR_URL"
+        env:
+          PR_URL: ${{ github.event.pull_request.html_url }}
+          # this token CANNOT be the default `GITHUB_TOKEN`, otherwise the merge of the PR will not trigger a build
+          GITHUB_TOKEN: ${{ secrets.token }}
+      - name: Auto-approve patch Dependabot PRs
+        if: ${{ steps.metadata.outputs.update-type == 'version-update:semver-patch' }}
+        run: gh pr review --approve "$PR_URL"
+        env:
+          PR_URL: ${{ github.event.pull_request.html_url }}
+          GITHUB_TOKEN: ${{ secrets.token }}
```

Comment:
can you move these two to the top level `env` given they are the same for both commands - in order to reduce duplication?

Reply:
I wanted to keep the specific comment for GITHUB_TOKEN on one of the actions, but ok

Reply:
(and this has been tested successfully without the `secrets: inherit` setting from the caller)